### PR TITLE
[NeuralChat] Enable bm25 sparse retrieval, enable retrieval then rerank pipeline

### PR DESCRIPTION
**Pull request by XuhuiRen**
## Type of Change

Fix the return format of retrievers to align the input of the reranker
delete unnecessary vector store retriever, the previous code do not need
enable retrieval then rerank with bge model (langchain do not have)
move the processing of the retrieved result to retrieval adapter
fix readme

### CODE REVIEW

File: intel_extension_for_transformers/langchain/retrievers/__init__.py [hunk 1]
```diff
@@ -15,5 +15,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from .vectorstore_retriever import VectorStoreRetriever
 from .child_parent_retriever import ChildParentRetriever
+from .bge_reranker import BgeReranker
```
Comment (hshen14) [inline intel_extension_for_transformers/langchain/retrievers/__init__.py:19]: can reranker apply to other embedding models except bge?

**Review by hshen14 — APPROVED** [review]

File: intel_extension_for_transformers/langchain/retrievers/__init__.py (hunk 1 above)
Comment (XuhuiRen) [reply intel_extension_for_transformers/langchain/retrievers/__init__.py:19]: it should work for all reranker models built with CrossEncoder structure

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/README.md [hunk 2]
```diff
@@ -167,7 +167,7 @@ We've chosen VectorStoreRetriever as the default retriever. This decision aligns
 Our approach ensures that users have access to versatile and effective retrieval tools, tailored to a variety of requirements and preferences within the system.
 
 ### VectorStoreRetriever
-We've maintained most of the retrieval behaviors consistent with langchain, but we've also introduced additional content processing steps. These enhancements are specifically designed to better meet our needs for source-based retrieval. The user can select this retriever by:
+We've maintained most of the retrieval behaviors consistent with langchain。 The user can select this retriever by:
```
Comment (lvliang-intel) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/README.md:170]: 。Typo, use English format

Comment (XuhuiRen) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/README.md]: fixed

**Review by lvliang-intel — APPROVED** [review]

**Comment by VincyZhang** [comment]
@XuhuiRen please resolve conflicts

**Pull request merged**